examples/tabular_image.py

```diff
@@ -0,0 +1,201 @@
+import argparse
+import os
+import os.path as osp
+from typing import List
+
+import torch
+import torch.nn.functional as F
+from PIL import Image
+from torch import Tensor
+from tqdm import tqdm
+
+from torch_frame import stype
+from torch_frame.config import ImageEmbedderConfig
+from torch_frame.data import DataLoader
+from torch_frame.datasets import DiamondImages
+from torch_frame.nn import (
+    EmbeddingEncoder,
+    FTTransformer,
+    LinearEmbeddingEncoder,
+    LinearEncoder,
+)
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--channels", type=int, default=256)
+parser.add_argument("--num_layers", type=int, default=4)
+parser.add_argument("--batch_size", type=int, default=512)
+parser.add_argument("--lr", type=float, default=0.0001)
+parser.add_argument("--epochs", type=int, default=30)
+parser.add_argument("--seed", type=int, default=0)
+parser.add_argument(
+    "--model",
+    type=str,
+    default="google/vit-base-patch16-224-in21k",
+    choices=[
+        "resnet18",
+        "google/vit-base-patch16-224-in21k",
+    ],
+)
+
+args = parser.parse_args()
+
+# ResNet-18 - Image Embedded
+# Best Val Acc: 0.2861, Best Test Acc: 0.2799
+# ViT - Image Embedded
+# Best Val Acc: 0.4221, Best Test Acc: 0.4269
+
+
+class ImageToEmbedding:
+    def __init__(self, model: str, device: torch.device):
+        if "resnet" in model:
+            from torchvision import transforms
+            model = torch.hub.load(
+                'pytorch/vision:v0.10.0',
+                model,
+                pretrained=True,
+            )
+            # Remove the last linear layer:
+            self.model = torch.nn.Sequential(
+                *(list(model.children())[:-1])).to(device)
+            # Preprocess transformations:
+            self.preprocess = transforms.Compose([
+                transforms.Resize(256),
+                transforms.CenterCrop(224),
+                transforms.ToTensor(),
+                transforms.Normalize(mean=[0.485, 0.456, 0.406],
+                                     std=[0.229, 0.224, 0.225]),
+            ])
+        elif "vit" in model:
+            from transformers import AutoImageProcessor, ViTModel
+            self.preprocess = AutoImageProcessor.from_pretrained(model)
+            self.model = ViTModel.from_pretrained(model).to(device)
+        self.model.eval()
+        self.device = device
+
+    def forward_retrieve(self, path_to_images: list[str]) -> list[Image]:
+        images: list[Image] = []
+        for path_to_image in path_to_images:
+            try:
+                image = Image.open(path_to_image)
+            except Exception:
+                # There is one image does not exist, thus create a black one:
+                image = Image.new("RGB", (600, 471))
```

Comment:
what's (600, 471) here? Does it make sense to just add this as the default behavior (and print out some warnings if image is not found)?